hnsendpoint.go

```diff
@@ -192,18 +192,22 @@ func (endpoint *HNSEndpoint) ContainerHotDetach(containerID string) error {
 	return modifyNetworkEndpoint(containerID, endpoint.Id, Remove)
 }
 
-// ApplyACLPolicy applies Acl Policy on the Endpoint
-func (endpoint *HNSEndpoint) ApplyACLPolicy(policy *ACLPolicy) error {
+// ApplyPolicy applies a set of ACL Policies on the Endpoint. This will
+// overwrite any existing policies which were previously applied
+func (endpoint *HNSEndpoint) ApplyACLPolicy(policies []*ACLPolicy) error {
 	operation := "ApplyACLPolicy"
 	title := "HCSShim::HNSEndpoint::" + operation
 	logrus.Debugf(title+" id=%s", endpoint.Id)
 
-	jsonString, err := json.Marshal(policy)
-	if err != nil {
-		return err
+	for _, policy := range policies {
+		jsonString, err := json.Marshal(policy)
+		if err != nil {
+			return err
+		}
+		endpoint.Policies = append(endpoint.Policies, jsonString)

```

Comment:
This looks like it only appends to existing policies, but the function doc above states that it overwrites. Am I missing something?

Reply:
Yes, thanks - the comment is from another iteration which we are working on (non-rs3 aligned). For now, this will be appending to existing policies as you said, so I've removed that comment.

Reply:
Should there be a way to clear the current ACL policies on an endpoint exposed separately then? Looks like this can be done my manually clearing the `endpoint.Policies` and calling `endpoint.Update`, but I don't think that is an ideal pattern.

Reply:
That's correct, that would be the workaround for a caller to remove (any kind of policy, not just ACL) for RS3. There are a few important changes to the policy model (especially for ACL) coming in the next release, so for now we are not planning to churn or add any more policy specific functions. This PR is an exception as we have a caller who needs to apply multiple ACLs for RS3 and without this change it's very painful for them.

Reply:
Understood. We do need to be careful with versioning here, as hcsshim is not tied to a particular release and is expected to work (and continue to work) on RS1 through current release.